src/actuator/ActuatorBladeDistributor.C

```diff
@@ -0,0 +1,120 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+#include <actuator/ActuatorBladeDistributor.h>
+#include <actuator/ActuatorBulkSimple.h>
+#include <NaluEnv.h>
+#ifdef NALU_USES_OPENFAST
+#include <actuator/ActuatorBulkFAST.h>
+#include <actuator/UtilitiesActuator.h>
+#endif
+
+namespace sierra {
+namespace nalu {
+
+bool
+blade_belongs_on_this_rank(
+  int numBladesTotal, int globBladeNum, int numRanks, int rank)
+{
+  const int div = numBladesTotal / numRanks;
+  const bool isInDivisionIncrement =
+    globBladeNum >= (div * rank) && globBladeNum < div * (rank + 1);
+  const bool isInRemainderIncrement = (globBladeNum - div * numRanks) == rank;
+  ThrowAssert(isInDivisionIncrement && isInRemainderIncrement);
+  return isInDivisionIncrement || isInRemainderIncrement;
+}
+
+std::vector<BladeDistributionInfo>
+compute_blade_distributions(const ActuatorMeta& actMeta, ActuatorBulk& actBulk)
+{
+  std::vector<BladeDistributionInfo> results;
+  const int rank = NaluEnv::self().parallel_rank();
+
+  switch (actMeta.actuatorType_) {
+  case (ActuatorType::ActLineSimpleNGP): {
+    auto actMetaSimp = dynamic_cast<const ActuatorMetaSimple&>(actMeta);
+    // one blade per processor for this case, but we could change this
+    if (!actMeta.entityFLLC_(actBulk.localTurbineId_))
+      break;
+    if (rank == actBulk.localTurbineId_) {
+      const int iBlade = actBulk.localTurbineId_;
+      const int offset = actBulk.turbIdOffset_.h_view(iBlade);
+      const int nPoints = actMetaSimp.num_force_pts_blade_.h_view(iBlade);
+      const int nNeighbor = actMetaSimp.numNearestPointsFllcInt_.h_view(iBlade);
```

Comment:
@tonyinme it's defined here. This is zero unless users specify it in the input deck. So the default is to integrate the full blade.